File: 477. Total Hamming Distance/477. Total Hamming Distance.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var totalHammingDistance = function(nums) {
    var res = 0;
    for(var i = 0; i < 32; i++) {
        var one = 0;
        var zero = 0;
        nums.forEach(function(val) {
            if((val >> i & 1)===0){
                zero++
            } else {
                one++
            }
        })
         res += zero * one;
    }
    return res;
}; 

File: 477. Total Hamming Distance/477. Total Hamming Distance.rb
# @param {Integer[]} nums
# @return {Integer}
def total_hamming_distance(nums)
	res = 0
	for i in 0..31 do
		zero , one = 0 , 0
		nums.each do |v|
			if (v >> i & 1) == 0
				zero += 1
			else
				one += 1
			end
		end
		res += zero * one
	end
	res
end
